### PR TITLE
Add rejected occurance

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Add the RejectedOccurrence model to database.
- Add the RejectedOccurrenceNode to GraphQL schema. Link node to RecurringReservationsNode.
- Add tests for the new GraphQL schema.
- Removes all Swedish translations, since they are not implemented yet. 
- Refactor the translation checker.

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3369][TILA]

[TILA]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3369

[TILA-3369]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/graphql/types/rejected_occurrence/filtersets.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            "__application_round"
+        )
+    )
+    reservation_unit = IntChoiceFilter(field_name="recurring_reservation__reservation_unit")
+    unit = IntChoiceFilter(field_name="recurring_reservation__reservation_unit__unit")
+    text_search = django_filters.CharFilter(method="filter_text_search")
+
+    class Meta:
+        model = RejectedOccurrence
+        order_by = [
```
Comment (vergama) [inline api/graphql/types/rejected_occurrence/filtersets.py:37]: Let's add `rejection_reason` as well, it was missing from the Jira ticket.

**Review by ranta — APPROVED** [review]

File: locale/sv/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -2834,6 +2834,30 @@ msgstr ""
...
+#, fuzzy
+#| msgid "Reservations begin date must be before reservations end date."
+msgid "Overlapping reservations"
+msgstr "Varausten alkupäivän täytyy olla ennen niiden loppupäivää."
+
+#: reservations/choices.py
+#, fuzzy
+#| msgid "Reservation Unit Options"
+msgid "Reservation unit closed"
+msgstr "Varausyksikkö-vaihtoehdot"
```
[the remark this answers is not in the document]
Comment (matti-lamppu) [reply locale/sv/LC_MESSAGES/django.po:2851]: This is the Swedish translation file. I'll remove all Swedish translation stuff for now, we can add them back later.

File: locale/sv/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -2834,6 +2834,30 @@ msgstr ""
...
 
+#: reservations/choices.py
+msgid "Interval not allowed"
+msgstr ""
+
+#: reservations/choices.py
+#, fuzzy
+#| msgid "Reservations begin date must be before reservations end date."
+msgid "Overlapping reservations"
+msgstr "Varausten alkupäivän täytyy olla ennen niiden loppupäivää."
```
[the remark this answers is not in the document]
Comment (matti-lamppu) [reply locale/sv/LC_MESSAGES/django.po:2845]: This is the Swedish translation file. I'll remove all Swedish translation stuff for now, we can add them back later.

File: api/graphql/types/rejected_occurrence/filtersets.py (hunk 1 above)
Comment (matti-lamppu) [reply api/graphql/types/rejected_occurrence/filtersets.py:37]: What should be the order when ordering by `rejection_reason`?

Comment (matti-lamppu) [reply api/graphql/types/rejected_occurrence/filtersets.py:37]: Implemented ordering (asc):
- `INTERVAL_NOT_ALLOWED`
- `OVERLAPPING_RESERVATIONS`
- `RESERVATION_UNIT_CLOSED`

**Review by vergama — APPROVED** [review]

**Comment by matti-lamppu** [comment]
Removed the Swedish translation file and all related functionality. Swedish translations can be implemented later if needed.

**Comment by matti-lamppu** [comment]
Merged #1220, #1216, and #1212 to this PR so that they can be merged to main together.

**Pull request merged**